### PR TITLE
Better partial iso generator

### PR DESCRIPTION
**Pull request by mbrandonw**
Some of our nested routes were getting pretty gnarly, so by extension `PartialIso` directly we can leverage type inference!

### CODE REVIEW

File: .sourcery-templates/DerivePartialIsos.stencil [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+{% for enum in types.implementing.DerivePartialIsos|enum %}
+  {% for case in enum.cases %}
+
+    {% if case.hasAssociatedValue %}
+
+      extension PartialIso where A == (
+        {% for value in case.associatedValues %}
+          {% if value.type %}
+            {{ value.type.globalName }}{% if value.typeName.isOptional %}?{% endif %}
```
Comment (mbrandonw) [inline .sourcery-templates/DerivePartialIsos.stencil]: there's a lot of hacky stuff happening in this stencil cause apparently sourcery and/or stencil doesnt have great nested type support. see more here https://github.com/krzysztofzablocki/Sourcery/issues/398

File: Sources/PointFree/Routers/GitHubRoutes.swift [hunk 2]
```diff
@@ -19,25 +19,25 @@ public enum GitHubRoute: DerivePartialIsos {
 
 let gitHubRouter = [
 
-  GitHubRoute.iso.authorize
+  PartialIso.authorize
```
Comment (mbrandonw) [inline Sources/PointFree/Routers/GitHubRoutes.swift]: for some strange reason I could not omit this `PartialIso` ... it's the only one that needed it!

File: Sources/PointFree/Routers/PointFreeRoutes.swift [hunk 3]
```diff
@@ -92,135 +92,127 @@ public enum Route: DerivePartialIsos {
...
+  .account <<< .subscription <<< .downgrade <<< .update
     <¢> post %> lit("account") %> lit("subscription") %> lit("downgrade") <% end,
 
-  Route.iso.account <<< Route.Account.iso.subscription <<< Route.Account.Subscription.iso.reactivate
+  .account <<< .subscription <<< .reactivate
     <¢> post %> lit("account") %> lit("subscription") %> lit("reactivate") <% end,
 
-  Route.iso.account <<< Route.Account.iso.subscription <<< Route.Account.Subscription.iso.upgrade
-    <<< Route.Account.Subscription.Upgrade.iso.show
+  .account <<< .subscription <<< .upgrade <<< .show
```
Comment (mbrandonw) [inline Sources/PointFree/Routers/PointFreeRoutes.swift]: much better

File: Tests/LinuxMain.swift [hunk 4]
```diff
@@ -29,14 +29,37 @@ extension AuthTests {
 extension CancelTests {
   static var allTests: [(String, (CancelTests) -> () throws -> Void)] = [
     ("testConfirmCancel", testConfirmCancel),
-    ("testCancel", testCancel)
+    ("testConfirmCancelLoggedOut", testConfirmCancelLoggedOut),
```
Comment (mbrandonw) [inline Tests/LinuxMain.swift]: all your tests weren't running on linux!

File: Sources/PointFree/Routers/GitHubRoutes.swift (hunk 2 above)
Comment (stephencelis) [reply Sources/PointFree/Routers/GitHubRoutes.swift]: You need one to hint the compiler! I think you could hint by giving `gitHubRouter` an explicit type, too.

File: Sources/PointFree/Routers/PointFreeRoutes.swift [hunk 5]
```diff
@@ -92,135 +92,127 @@ public enum Route: DerivePartialIsos {
 
 private let routers: [Router<Route>] = [
 
-  Route.iso.about
+  .about
     <¢> get %> lit("about") <% end,
 
-  Route.iso.account <<< Route.Account.iso.confirmEmailChange
+  .account <<< .confirmEmailChange
```
Comment (stephencelis) [inline Sources/PointFree/Routers/PointFreeRoutes.swift]: Beaut!

File: Tests/LinuxMain.swift (hunk 4 above)
Comment (stephencelis) [reply Tests/LinuxMain.swift]: 🙈

File: Sources/PointFree/Routers/GitHubRoutes.swift (hunk 2 above)
Comment (mbrandonw) [reply Sources/PointFree/Routers/GitHubRoutes.swift]: but somehow it works for `TwitterRoute`!

**Pull request merged**